backend/controllers/todoControllers.js

```diff
@@ -0,0 +1,72 @@
+import { getDataBase } from '../db/index.js'
+import { validateValue } from '../helpers.js'
+
+const db = getDataBase()
+
+export const getAllTodos = async (req, res) => {
+  try {
+    const allTodos = await db.findAsync({})
+    res.send(allTodos)
+  } catch (error) {
+    res.status(500).send({error: "Failed to receive tasks"})
+  }
+  
+}
+
+export const addTask = async (req, res) => {
+  const body = req.body
+  
+  try {
+    await validateValue(body)
+    const task = await db.insertAsync({
+      text: body.text,
+      is_checked: body.is_checked,
+    })
+
+    res.send(task)
+  } catch (error) {
+    res.status(400).send({ error: "Missing required fields" })
+  }
+
+}
+
+export const deleteTask = async (req, res) =>  {
+  const taskId = req.params.id
+
+  const existingTask = await db.findOneAsync({ _id: taskId })
+
+  if (!existingTask) {
+    return res.status(404).send({ error: "Task not found" })
+  }
+
+  try {
+    await db.removeAsync({ _id: taskId })
+    res.send({ message: "Task deleted successfully" })
+  } catch (error) {
+    res.status(500).send({ error: "Failed to delete task" })
+  }
+}
+
+export const updateTask = async (req, res) => {
+  const taskId = req.params.id;
+  const updatedTaskData = req.body;
+
+  const existingTask = await db.findOneAsync({ _id: taskId });
+  if (!existingTask) {
+    return res.status(404).send({ error: "Task not found" });
+  }
+
+  try {
+    await validateValue(updatedTaskData)
+    await db.updateAsync({ _id: taskId }, { $set: updatedTaskData });
+
+    const updatedTask = { ...existingTask, ...updatedTaskData };
```

Comment:
Instead of constructing the `updatedTask` manually, you can use `returnUpdatedDocs` option in `update` which will return the updated documents automatically